src/Provider/OldOrderProvider.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusOrderEditPlugin\Provider;
+
+use Sylius\Component\Core\Inventory\Operator\OrderInventoryOperatorInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class OldOrderProvider
+{
+    public function __construct(
+        private OrderRepositoryInterface $orderRepository,
+        private OrderInventoryOperatorInterface $orderInventoryOperator,
+    ) {
+    }
+
+    public function provide(int $orderId): OrderInterface
+    {
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->find($orderId);
+        Assert::isInstanceOf($order, OrderInterface::class);
+
+        $this->orderInventoryOperator->cancel($order);
```

Comment:
Taking the name of the service into consideration I wouldn't expect this call to happen here

Reply:
I totally agree with that comment - but, as it's mentioned in the PR description, this is just a refactor of the current logic, so we would be able to cover such a case later. And that's also the reason for having the separate service for that, so we can prepare the order to be updated in one place :dancer:

Anyway, I agree there is a naming issue here. Will try to figure out something better 🖖